packages/frontend/src/pages/UserDetails/UserDetailsPage.tsx

```diff
@@ -0,0 +1,88 @@
+import { faUserGroup } from '@fortawesome/free-solid-svg-icons';
+import BreadCrumb from '@/components/BreadCrumb';
+import BackLink from '@/navigation/BackLink';
+import { SingleUser, SingleUserParams, useAdminUsers } from '@/model/users';
+import { useRecoilValue } from 'recoil';
+import { useParams } from 'react-router-dom';
+import { formatIsoDateUTC } from '@/utils/date';
+import { shortenEthAddress } from '@/utils/index';
+import { UserDto, UserRole } from 'api/dist/user/types';
+import { toast } from 'react-hot-toast';
+
+const roles = [UserRole.ADMIN, UserRole.FORWARDER, UserRole.QUANTIFIER];
+
+const UserDetailsPage = (): JSX.Element => {
+  const { userId } = useParams<SingleUserParams>();
+  const user = useRecoilValue(SingleUser({ userId }));
+  const { addRole, removeRole } = useAdminUsers();
+
+  const handleRole = (role: UserRole, user: UserDto): void => {
+    let resp;
+    const isRemove = user.roles.includes(role);
+    async (): Promise<void> => {
+      if (isRemove) {
+        resp = await removeRole(user._id, role);
+      } else {
+        resp = await addRole(user._id, role);
+      }
+      if (resp?.status === 200) {
+        toast.success(`Role ${isRemove ? 'removed' : 'added'} successfully!`);
+      }
+    };
+  };
+
+  return (
+    <div className="max-w-2xl mx-auto">
+      <BreadCrumb name="User details" icon={faUserGroup} />
+      <BackLink to="/pool" />
+      <div className="praise-box flex flex-col gap-2">
+        <span>User identity</span>
+        {!!user && (
+          <>
+            <span className="text-xl font-bold">
+              {shortenEthAddress(user.ethereumAddress || '')}
+            </span>
+            <span>Created: {formatIsoDateUTC(user.createdAt)}</span>
+            <span>Last updated: {formatIsoDateUTC(user.updatedAt)}</span>
+          </>
+        )}
+      </div>
+      <div className="praise-box flex flex-col gap-2">
+        <span>Linked Discord identities</span>
+        {user?.accounts?.map((account, index) => (
+          <>
+            <span>User account #{index + 1}</span>
```

Comment:
This text "User Account #x" is not necessary.

Reply:
Yes, a user can only have one linked Discord and/or Telegram account. Unless you are running locally for development. This happens when you start with a fresh db with dummy data and then activate your account using the bot. Two Discord user accounts, should not be possible right @mattyg. I created a new issue #398.

Reply:
> Yes, a user can only have one linked Discord and/or Telegram account. Unless you are running locally for development. This happens when you start with a fresh db with dummy data and then activate your account using the bot. Two Discord user accounts, should not be possible right @mattyg. I created a new issue #398.

To be clear though -- this is not an issue with this PR, and requires no changes.